lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java

```diff
@@ -86,9 +91,10 @@ static final class State implements Serializable {
 		private final TreeMap<String, String> userData;
 		private final TreeMap<String, Object> editorConfigOverride;
 		private final String version;
+		private final String editorConfigPath;
```

Comment:
This field needs to be `FileSignature editorConfig`, you can create it like this

https://github.com/diffplug/spotless/blob/86404f1a1708b9fcb97ef8d7c16a43bec9bc7e06/lib/src/main/java/com/diffplug/spotless/FileSignature.java#L52

and then access it like this

https://github.com/diffplug/spotless/blob/86404f1a1708b9fcb97ef8d7c16a43bec9bc7e06/lib/src/main/java/com/diffplug/spotless/FileSignature.java#L101-L102

The reason is that if the content of `.editorconfig` file changes, the up-to-date status of the tasks must change too. FileSignature handles this.

Reply:
I resisted to pass FileSignature below `KtlintFormatterFunc` to skip adding an additional dependency